test/Table_Tests/src/Common_Table_Operations/Join/Cross_Join_Spec.enso

```diff
@@ -159,5 +158,5 @@ spec setup =
             Test.specify "Cross join via a direct call to .join should not allow join conditions" <|
                 t1 = table_builder [["X", [1, 2]], ["Y", [4, 5]]]
                 t2 = table_builder [["Z", ['a', 'b']], ["W", ['c', 'd']]]
-                j = t1.join t2 join_kind=Join_Kind.Cross on=[Join_Condition.Equals 'X' 'Z']
+                j = t1.join t2 join_kind=Join_Kind_Cross.Cross on=[Join_Condition.Equals 'X' 'Z']
                 j . should_fail_with Illegal_Argument
```

Comment:
This test should be amended, I think `Cross` join thru call to `join` should not be allowed at all.

These are still separate methods. Even if we share most of the implementation, we may change that in the future - so the API should not expose this sharing at all.

```
            Test.specify "Cross join is not possible via call to .join" <|
                t1 = table_builder [["X", [1, 2]], ["Y", [4, 5]]]
                t2 = table_builder [["Z", ['a', 'b']], ["W", ['c', 'd']]]
                Test.expect_panic_with (t1.join t2 join_kind=Join_Kind_Cross.Cross on=[]) Type_Error      
```

Reply:
Done.